astropy/coordinates/builtin_frames/cirs_observed_transforms.py

```diff
@@ -86,17 +92,32 @@ def altaz_to_cirs(altaz_coo, cirs_frame):
                          altaz_coo.obswl.value)
 
     # the 'A' indicates zen/az inputs
-    cirs_ra, cirs_dec = erfa.atoiq('A', az, zen, astrom)
+    cirs_ra, cirs_dec = erfa.atoiq('A', az, zen, astrom)*u.radian
 
     if isinstance(altaz_coo.data, UnitSphericalRepresentation):
         distance = None
     else:
-        #now we get the distance as just the cartesian
-        #distance from the earth location to the coordinate location
-        distance = altaz_coo.separation_3d(altaz_coo.location.itrs)
+        locitrs = altaz_coo.location.get_itrs(altaz_coo.obstime)
+
+        # To compute the distance in a way that is reversable with cirs_to_altaz
```

Comment:
This block is in particular need of review/somebody else considering it.  The geometry is rather confusing, but most importantly, I'm not certain how numerically stable this solution to the SSA trig problem is. So if anyone else knows a trick for this particular geometric calculation I'd love to hear it.